test/Microsoft.AspNet.Mvc.Core.Test/DefaultAssemblyProviderTests.cs

```diff
@@ -98,6 +100,51 @@ public void CandidateAssemblies_ReturnsEmptySequenceWhenReferenceAssembliesIsNul
             Assert.Empty(nullProviderCandidates.Select(a => a.Name));
         }
 
+        // This test verifies DefaultAssemblyProvider.ReferenceAssemblies returns the assembly list that we intended.
+        // If ReferenceAssemblies needs to be changed, the expected list in this test should be updated together.
+        [Fact]
+        public void ReferenceAssemblies_ReturnsExpectedReferenceAssemblies()
```

Comment:
Suppose someone inadvertently adds another assembly to the list. Their immediate next step is to just "fix" the unit test and the product is then broken. As such, I think a big giant warning comment in the product source code would yield a better result.

If there's a way to programmatically generate the list, or somehow verify that the list doesn't include anything that can't possibly be correct, that would be great. But comparing one static list to another static list doesn't do much.